src/engine/lua.h

```diff
@@ -101,4 +101,27 @@ static const struct luaL_Reg mscLuaLib[] = {
 }  // namespace engine
 }  // namespace modsecurity
 
+#if !defined LUA_VERSION_NUM || LUA_VERSION_NUM == 501
+/*
+** Adapted from Lua 5.2.0
+*/
+#define LUA_OK 0
+
+static void luaL_setfuncs(lua_State *L, const luaL_Reg *l, int nup) {
+    luaL_checkstack(L, nup + 1, "too many upvalues");
+
+    for (; l->name != NULL; l++) {  /* fill the table with given functions */
+        int i;
+        lua_pushstring(L, l->name);
+        for (i = 0; i < nup; i++)  /* copy upvalues to the top */
+          lua_pushvalue(L, -(nup + 1));
+        lua_pushcclosure(L, l->func, nup);  /* closure with those upvalues */
+        lua_settable(L, -(nup + 3));
+    }
+
+    lua_pop(L, nup);  /* remove upvalues */
+}
+#endif
```

Comment:
_Maybe_ the intent is to eventually promote support for LuaJIT with this PR (totally speculating here), but if so, and if I recall correctly, some versions of LuaJIT define `LUA_VERSION_NUM` as `501` and implement `luaL_setfuncs`. Should we follow Mike's advice here? See:

https://www.freelists.org/post/luajit/ANN-LuaJIT210beta3,3

Reply:
If this isn't a goal nor a long-term goal of this PR, then feel free to ignore this!

Reply:
Hey @thibaultcha, the goal here is just to get support for 5.1 introduced. Trying to integrate with OpenResty was a bit of an exercise (see #1809), and there's no clean way to do it at this point (it was also the initial impetus for this PR). Eventually LuaJIT support might be nice, but since a number of distros only provide Lua lib packages for 5.1 out of the box, getting 5.1 supported is the sole goal here. As such I'd rather not introduce new symbols unnecessarily and increase the scope of this change.

If there's a large push for LuaJIT support here, perhaps we can revisit in the future, but at this point I do not see much value for that.

Reply:
All good :+1:

Reply:
There's an attempt to support LuaJIT on some distros at https://github.com/SpiderLabs/ModSecurity/pull/1854. You might want to give it a try to see if it works for you @thibaultcha.